### PR TITLE
fix malformatted number to fit eth hex format (#422)

### PR DESCRIPTION
**Pull request by glihm**
# Pull Request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

`eth_` endpoints returning numbers are displayed in decimal. 

Issue Number: #422 

# What is the new behavior?

`eth_` endpoints returning numbers are now displayed as hexadecimal string.

# Does this introduce a breaking change?

- [ ] Yes
- [X] No

# Other information

This PR is not conflicted with #416 and can be merged to achieve some issues listed in #422.

### CODE REVIEW

File: crates/beerus-rpc/src/lib.rs [hunk 1]
```diff
@@ -181,7 +184,7 @@ impl BeerusRpcServer for BeerusRpc {
             .get_gas_price()
             .await
             .map_err(|_| Error::from(BeerusApiError::ContractError))?;
-        Ok(gas_price.to_string())
+        Ok(format!("0x{gas_price:x}"))
```
Comment (phklive) [inline crates/beerus-rpc/src/lib.rs:187]: We need to check if all methods that should return hex considering the Ethereum rpc specs: 

Indeed return hex, and does that should not, do not, so we are 100% compliant.

File: crates/beerus-rpc/src/lib.rs [hunk 2]
```diff
@@ -69,7 +69,7 @@ impl BeerusRpcServer for BeerusRpc {
             .await
             .map_err(|_| Error::from(BeerusApiError::ContractError))?;
 
-        Ok(balance.to_string())
+        Ok(format!("0x{balance:x}"))
```
[the remark this answers is not in the document]
Comment (glihm) [reply crates/beerus-rpc/src/lib.rs:72]: I started doing like this, but I didn't want to add a utils crate. I can add a module utils in EPC, which yes you are right should be more organized.

File: crates/beerus-rpc/src/utils.rs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+use ethers::types::U256;
```
Comment (phklive) [inline crates/beerus-rpc/src/utils.rs:1]: I think that the utils file should go in the beerus-core utils folder: 
https://github.com/keep-starknet-strange/beerus/blob/main/crates/beerus-core/src/ethers_helper.rs

**Review by phklive — COMMENTED** [review]
Potential modification of placement of utils fn. 

Rest lgtm.

File: crates/beerus-rpc/src/utils.rs (hunk 3 above)
Comment (glihm) [reply crates/beerus-rpc/src/utils.rs:1]: It's a very generic function of formatting we may use it for other RPC purposes.

What if tomorrow some starknet output must be hexadecimal?

But if it's sound better to you, let me know and I'll move it to the place you point out.

Comment (glihm) [reply crates/beerus-rpc/src/utils.rs:1]: Or perhaps changing to a generic function or macro.

**Pull request merged**